### PR TITLE
mark orders as taken in db

### LINKED ISSUE #261: Update order as taken in orderbook
After matching an order we should update the db so that the orders are not matched again. 

_Originally posted by @holzeis in https://github.com/get10101/10101/pull/248#discussion_r1135364628_

### PR DESCRIPTION
**Pull request by 1010Tom**
resolves #261

### CODE REVIEW

File: coordinator/src/orderbook/routes.rs [hunk 1]
```diff
@@ -92,17 +93,37 @@ pub async fn post_order(
...
+            .iter()
+            .map(|m| m.order_id)
+            .collect();
+
+        orders_to_set_taken.append(&mut order_ids);
+
         notify_traders(matched_orders, authenticated_users.clone()).await;
+
+        for order_id in orders_to_set_taken {
+            if let Err(err) = db::orders::update(&mut conn, order_id, true) {
```
Comment (holzeis) [inline coordinator/src/orderbook/routes.rs:122]: nit: the method name could be more clear to what this is doing. I would have expected an update method to update the whole order and not just the taken flag. How about `db::orders::taken(&mut conn, order_id, true)`?

**Review by holzeis — APPROVED** [review]
LGTM 👍

File: coordinator/src/orderbook/routes.rs (hunk 1 above)
Comment (1010Tom) [reply coordinator/src/orderbook/routes.rs]: done

**Pull request merged**